lib/rules/require-semicolons.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Requires semicolon after:
+ *
+ * * var declaration
+ * * expression statement
+ * * return
+ * * throw
+ * * break
+ * * continue
+ * * do-while
+ *
+ * Type: `Boolean`
+ *
+ * Values: `true`
+ *
+ * #### Example
+ *
+ * ```js
+ * "requireSemicolons": true
+ * ```
+ *
+ * ##### Valid
+ *
+ * ```js
+ * var a = 1;
+ * ```
+ *
+ * ##### Invalid
+ *
+ * ```js
+ * var a = 1
+ * ```
+*/
+
+var assert = require('assert');
+
+module.exports = function() {};
+
+module.exports.prototype = {
+
+    configure: function(requireSemicolons) {
+        assert(
+            typeof requireSemicolons === 'boolean',
+            'requireSemicolons option requires boolean value'
+        );
+        assert(
+            requireSemicolons === true,
+            'requireSemicolons option requires true value or should be removed'
+        );
+    },
+
+    getOptionName: function() {
+        return 'requireSemicolons';
+    },
+
+    check: function(file, errors) {
+        file.iterateNodesByType([
+            'VariableDeclaration',
+            'ExpressionStatement',
+            'DoWhileStatement',
+            'ReturnStatement',
+            'ThrowStatement',
+            'BreakStatement',
+            'ContinueStatement',
+            'DebuggerStatement'
+        ], function(node) {
+            // ignore variable declaration inside for and for-in
+            if (node.type === 'VariableDeclaration') {
+                if ((node.parentNode.type === 'ForInStatement' && node.parentNode.left === node) ||
```

Comment:
I added `ForOfStatement` support and basic ES6 support as well (tests for `let` and `const` was added). But I'm not sure ES6 is fully supported still (even by `Esprima`).